include/aws/common/json.h

```diff
@@ -166,6 +166,31 @@ bool aws_json_value_has_key(const struct aws_json_value *object, struct aws_byte
  */
 AWS_COMMON_API
 int aws_json_value_remove_from_object(struct aws_json_value *object, struct aws_byte_cursor key);
+
+/**
+ * @brief callback for iterating keys of an object
+ * return false to exit early
+ */
+typedef bool(aws_json_on_member_encountered_const_fn)(
+    const struct aws_byte_cursor *key,
+    const struct aws_json_value *value,
+    void *user_data);
+
+/**
+ * @brief iterates though fields of the object.
```

Comment:
trivial: "iterates though fields" -> "iterates through members"

also, decent spell checker plugin for vscode: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker